pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -22,17 +37,158 @@ func init() {
+	if !ok {
 		return nil, nil, nil, false, nil
 	}
-	return nil, nil, nil, false, errors.UnimplementedError(issuelink.IssueLink{Detail: "plan hook to be implemented"}, "plan hook to be implemented")
+
+	exprEval := p.ExprEvaluator("LOGICAL REPLICATION STREAM")
+
+	from, err := exprEval.String(ctx, stmt.PGURL)
+	if err != nil {
+		return nil, nil, nil, false, err
+	}
+
+	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) (err error) {
+		defer func() {
+			if err == nil {
+				telemetry.Count("logical_replication_stream.started")
+			}
+		}()
+		ctx, span := tracing.ChildSpan(ctx, stmt.StatementTag())
+		defer span.Finish()
+
+		if !p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.V24_2) {
+			return pgerror.New(pgcode.FeatureNotSupported,
+				"replication job not supported before V24.2")
+		}
+
+		if err := utilccl.CheckEnterpriseEnabled(
+			p.ExecCfg().Settings,
+			"CREATE LOGICAL REPLICATION",
+		); err != nil {
+			return err
+		}
+
+		if !stmt.Options.IsDefault() {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication stream options are not yet supported")
+		}
+		if stmt.On.Database != "" {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication streams on databases are unsupported")
+		}
+		if len(stmt.On.Tables) != len(stmt.Into.Tables) {
+			return pgerror.New(pgcode.InvalidParameterValue, "the same number of source and destination tables must be specified")
+		}
+
+		streamAddress := crosscluster.StreamAddress(from)
+		streamURL, err := streamAddress.URL()
+		if err != nil {
+			return err
+		}
+		streamAddress = crosscluster.StreamAddress(streamURL.String())
+
+		client, err := streamclient.NewStreamClient(ctx, streamAddress, p.ExecCfg().InternalDB)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			_ = client.Close(ctx)
+		}()
+
+		if err := client.Dial(ctx); err != nil {
+			return err
+		}
+
+		var targetsDescription string
+		srcTableNames := make([]string, len(stmt.On.Tables))
+
+		repPairs := make([]jobspb.LogicalReplicationDetails_ReplicationPair, len(stmt.Into.Tables))
+		for i := range stmt.On.Tables {
+
+			dstObjName, err := stmt.Into.Tables[i].ToUnresolvedObjectName(tree.NoAnnotation)
+			if err != nil {
+				return err
+			}
+			dstTableName := dstObjName.ToTableName()
+			prefix, td, err := resolver.ResolveMutableExistingTableObject(ctx, p, &dstTableName, true, tree.ResolveRequireTableDesc)
+			if err != nil {
+				return err
+			}
+
+			repPairs[i].DstDescriptorID = int32(td.GetID())
+
+			tbNameWithSchema := tree.MakeTableNameWithSchema(
+				tree.Name(prefix.Database.GetName()),
+				tree.Name(prefix.Schema.GetName()),
+				tree.Name(td.GetName()),
+			)
+
+			srcTableNames[i] = stmt.On.Tables[i].String()
+
+			if i == 0 {
+				targetsDescription = tbNameWithSchema.FQString()
+			} else if i < 5 {
+				targetsDescription += ", " + tbNameWithSchema.FQString()
+			} else if i == 5 {
```

Comment:
Is this just arbitrarily chosen?

Reply:
Yeah, we don’t like long job descriptions

Reply:
Actually just remembered that I changed `SHOW JOBS` to truncate long descriptions to 100 chars at display time so I removed the 5 cap here.